### PR TITLE
Default file handler

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -1,4 +1,10 @@
 -------------------------------------------------------------------
+Mon Dec 21 13:29:54 UTC 2015 - jreidinger@suse.com
+
+- allow specify default file handler all models
```
Comment (mvidner) [inline CHANGELOG]: > allow specifying a default file handler for all models

File: spec/base_model_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+require "cfa/base_model"
+require "cfa/memory_file"
+
+describe CFA::BaseModel do
+  let(:parser) { CFA::AugeasParser.new("sudoers.lns") }
+  let(:path) { "/etc/sudoers" }
+  subject { CFA::BaseModel.new(parser, path)}
+
+  describe ".default_file_handler" do
+    it "returns object set by .default_file_handler" do
```
Comment (ancorgs) [inline spec/base_model_spec.rb]: NP: I'd say this block is redundant to the last one. I know it's looking to the same thing from both the setter and the getter, but a little bit too much for my taste. :-)

**Comment by ancorgs** [comment]
Just a NP (and English correction by Martin). In sort, LGTM.

File: spec/base_model_spec.rb (hunk 2 above)
Comment (jreidinger) [inline spec/base_model_spec.rb]: I know, better idea how to test it? it is even very trivial part of code, so I am not sure if it deserve test. It is basically cattr_accessible just with default value ( rails can do not it, but ruby probably not ).

File: lib/cfa/base_model.rb [hunk 3]
```diff
@@ -12,9 +12,10 @@ class BaseModel
     # @param file_path [String] expected path passed to file_handler
     # @param file_handler [.read, .write] object, that can read/write string.
     #   It have to provide methods `string read(string)` and
-    #   `write(string, string)`. For example see {CFA::MemoryFile}
-    def initialize(parser, file_path, file_handler: File)
-      @file_handler = file_handler
+    #   `write(string, string)`. For example see {CFA::MemoryFile}. When nil
+    #   passed it use object from default_file_handler.
```
Comment (mvidner) [inline lib/cfa/base_model.rb]: A Yardoc link to the method + grammar fixes:
```
    # @param file_handler [.read, .write] an object able to read/write a string.
    #   It has to provide methods `string read(string)` and
    #   `write(string, string)`. For an example see {CFA::MemoryFile}.
    #   If unspecified or `nil`, {.default_file_handler} is asked.
```

**Pull request merged**